lazy_theta_star_planner/src/lazy_theta_star_planner.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "lazy_theta_star_planner/lazy_theta_star_planner.h"
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+
+namespace lazyThetaStarPlanner
+{
+void LazyThetaStarPlanner::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmapRos)
+{
+  planner_ = std::make_unique<lazyThetaStar::LazyThetaStar>();
+  node_ = parent;
+  name_ = name;
+  planner_->costmap_ = costmapRos->getCostmap();
+  globalFrame_ = costmapRos->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", how_many_corners_);
+
+  if (how_many_corners_ != 8) {
+    how_many_corners_ = 4;
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+      0.1));
+
+  node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".lethal_cost", rclcpp::ParameterValue(
+      127));
+
+  node_->get_parameter(name_ + ".lethal_cost", lethal_cost_);
+}
+
+void LazyThetaStarPlanner::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+void LazyThetaStarPlanner::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type lazyThetaStarPlanner",
+    name_.c_str());
+}
+
+nav_msgs::msg::Path LazyThetaStarPlanner::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  nav_msgs::msg::Path global_path;
+  std::vector<coords<world_pts>> path;
+
+  planner_->costmap_->worldToMap(start.pose.position.x, start.pose.position.y, src_[0], src_[1]);
+  planner_->src = {static_cast<int>(src_[0]), static_cast<int>(src_[1])};
```

Comment:
Rather than static casting, is it possible to use unsigned in in the src/dst in the planner itself?

Reply:
No, cause then it messes up when subtraction is done to calculate distance

Reply:
Ah ok